File: golang/client/commands/insertClientCommandHandler.go
package client

import (
	client "github.com/cqrs-govscsharp/golang/client/domain"
	commands "github.com/cqrs-govscsharp/golang/domain/commands"
	events "github.com/cqrs-govscsharp/golang/domain/events"
)

type InsertClientCommandHandler struct {
	commands.CommandHandler
	storage  client.ClientStorager
	eventBus events.EventBuser
}

func NewInsertClientCommandHandler(storage client.ClientStorager, eventBus events.EventBuser) InsertClientCommandHandler {
	return InsertClientCommandHandler{
		storage:  storage,
		eventBus: eventBus,
	}
}

func (ch InsertClientCommandHandler) CommandName() string {
	return "InsertClientCommand"
}

func (ch InsertClientCommandHandler) Handle(command commands.Commander) error {
	c := client.NewClient(command.(client.InsertClientCommand).ClientName())
	err := ch.storage.Add(c)

	if err != nil {
		return err
	}

	return ch.eventBus.Publish(c.PendingEvents()...)
}

File: csharp/src/CQRSService.Client/Commands/InsertClientCommandHandler.cs
using CQRSService.Client.CQRSService.Domain.Commands;
using CQRSService.Client.Storage;
using CQRSService.Domain.Commands;
using CQRSService.Domain.Events;
using MediatR;
using System.Threading;
using System.Threading.Tasks;

namespace CQRSService.Client.Commands
{
    public class InsertClientCommandHandler : ICommandHandler<InsertClientCommand>
    {
        private readonly IStorage<Client> _clientStorage;
        private readonly IEventBus _eventBus;

        public InsertClientCommandHandler(IStorage<Client> clientStorage, IEventBus eventBus)
        {
            _clientStorage = clientStorage;
            _eventBus = eventBus;
        }
        public async Task<Unit> Handle(InsertClientCommand request, CancellationToken cancellationToken)
        {
            var client = new Client(request.Name);
            await _clientStorage.Add(client);
            await _eventBus.Publish(client.PendingEvents.ToArray());
            return Unit.Value;
        }
    }
}
